src/librustc/mir/mod.rs

```diff
@@ -6,7 +6,9 @@
 
 use crate::hir::def::{CtorKind, Namespace};
 use crate::hir::def_id::DefId;
-use crate::hir::{self, InlineAsm as HirInlineAsm};
+// FIXME: check if we shouldn't create a new GeneratorKind struct for the MIR only or move the
```

Comment:
How would the struct look like / what would be the difference?

Reply:
There would be no difference to the GeneratorKind itself. That is a bit of a confusing FIXME comment. I thought that it may be a good idea to move its definition (the GeneratorKind struct) out of the HIR since it would now also be referenced by the MIR.